doc/manuals/devel/httpHeaders.md

```diff
@@ -0,0 +1,353 @@
+13) [Access-Control-Allow-Methods](#13-access-control-allow-methods)
+14) [Access-Control-Max-Age](#14-access-control-max-age)                                 
+15) [Access-Control-Expose-Headers](#15-access-control-expose-headers)
+16) [Allow](#16-allow)                                                                                    
+17) [Location](#17-location)                                                 
+18) [Ngsiv2-AttrsFormat](#18-ngsiv2-attrsformat)                                    
+19) [Fiware-Service](#19-fiware-service)
+20) [Fiware-Servicepath](#20-fiware-servicepath)
+21) [Fiware-Total-Count](#21-fiware-total-count)
+22) [Fiware-Correlator](#22-fiware-correlator)                                     
+
+23) [X-Auth-Token](#23-x-auth-token)
+
+
+
+# HTTP Headers used by Orion with their Description:
+
+
+
+## 1. Accept
+
+Accept HTTP Header is used in requests. 
+Accept HTTP Header is used with the request URL for accepting the response in well defined JSON format. 
+For example in request URL if we use:
+	
+    curl ... 'Accept: application/json'
+		 
+On this request we'll get the response data in well defined JSON format.
+
+The API response payloads in this specification are based on `application/json` and  (for attribute value 
+type operation) `text/plain` MIME types. Clients issuing HTTP requests with accept types different 
+than those will get a `406 Not Acceptable` error.
+
+[Top](#top)
+
+## 2. Content-Length
+
+This header is used by both requests and responses.
+Content-length HTTP header is a mandatory header in Orion responses.
+it defines the length of the request and response body in octets (8-bit bytes).
+Orion Context Broker expects always a Content-Length header in all client requests, otherwise the client will receive a `411 Length Required` response. 
+This is due to the way the underlying HTTP library (microhttpd) works.
+	
+    Content-Length: 34	
+
+[Top](#top)
+ 
+## 3. Content-Type 
+
+It is used to send the requests in well defined JSON format.
+
+    curl ... 'Content-Type: application/json'
+
+This request will send data in well defined JSON format.
+
+Specifically, if the resulting URL after applying the template is malformed, then no notification is sent.
+
+[Top](#top)
+ 
+## 4. Connection
+ 
+ This Header is used by both requests and responses.
+ It is used as Control options for the current connection and list of hop-by-hop response fields.
+	
+    Connection: Keep-Alive
+    	
+[Top](#top)	
+  
+## 5. Expect
+
+Expect HTTP Header is used in requests.
+It Indicates that behaviour of particular server are required by the client.
+It is typically used when sending a large request body. We expect the server to return back `100 Continue` HTTP status if it can handle the request, or `417 Expectation Failed` if not.
+
+    Expect: 100-continue
+	
+[Top](#top)	
+	
+## 6. Host
```

Comment:
It seems Orion doesn't actually use this header. In that case, I'd suggest to remove it from httpHeaders.h

What do you think?

Reply:
Host HTTP header is used in following files:
[rest/httpRequestSend.cpp#L375](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/httpRequestSend.cpp#L375)
[rest/rest.cpp#L504](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/rest.cpp#L504)
If we remove the header then it may affect the existing code.
Please provide your comment on this.

Reply:
With regards to [rest/httpRequestSend.cpp#L375](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/httpRequestSend.cpp#L375), you are right: CB explictely add the header base in outgoing requests.

However, with regards to [rest/rest.cpp#L504](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/rest.cpp#L504) that code just move the value of the hader to `headerP->host`. However, if we look for usages of that `headerP->host` field, it seems there isn't. Could you confirm? 

In that case, that logic could be removed and the code simplified.

Reply:
Yes, you are correct.

> With regards to [rest/httpRequestSend.cpp#L375](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/httpRequestSend.cpp#L375), you are right: CB explictely add the header base in outgoing requests.

So we should not remove `Host` HTTP header from `HttpHeaders.h` as it is being used in [rest/httpRequestSend.cpp#L375](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/httpRequestSend.cpp#L375) file.

> However, with regards to [rest/rest.cpp#L504](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/rest.cpp#L504) that code just move the value of the hader to `headerP->host`. However, if we look for usages of that `headerP->host` field, it seems there isn't. Could you confirm?
> 
> In that case, that logic could be removed and the code simplified.

So we should only remove the code from [rest/rest.cpp#L504](https://github.com/telefonicaid/fiware-orion/blob/master/src/lib/rest/rest.cpp#L504)??
Please provide your comment.

Reply:
> So we should not remove Host HTTP header from HttpHeaders.h as it is being used in rest/httpRequestSend.cpp#L375 file.

Correct.

> So we should only remove the code from rest/rest.cpp#L504??
Please provide your comment.

Correct. I'll do so, in a separte PR (I'll let you know, as reviewer). In this PR, let's focus in modification to documantation only.